tests/topotests/multicast_pim6_static_rp_topo1/test_multicast_pim6_static_rp1.py

```diff
@@ -0,0 +1,1317 @@
+    result = socat_send_mld_join(
+        tgen, "r0", "UDP6-RECV", GROUP_ADDRESS_1, intf, intf_ip
+    )
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("r1: Verify MLD groups")
+    dut = "r1"
+    intf_r1_r0 = TOPO["routers"]["r1"]["links"]["r0"]["interface"]
+    result = verify_mld_groups(tgen, dut, intf_r1_r0, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r5: Send multicast traffic for group ff08::1")
+    intf = TOPO["routers"]["r5"]["links"]["r3"]["interface"]
+    SOURCE_ADDRESS = TOPO["routers"]["r5"]["links"]["r3"]["ipv6"].split("/")[0]
+    result = socat_send_pim6_traffic(tgen, "r5", "UDP6-SEND", GROUP_ADDRESS_1, intf)
+    assert result is True, "Testcase {}: Failed Error: {}".format(tc_name, result)
+
+    step("r1: Verify (*, G) upstream IIF interface")
+    dut = "r1"
+    iif = TOPO["routers"]["r1"]["links"]["r2"]["interface"]
+    oif = TOPO["routers"]["r1"]["links"]["r0"]["interface"]
+    result = verify_upstream_iif(tgen, dut, iif, STAR, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r1: Verify (*, G) upstream join state and join timer")
+    result = verify_join_state_and_timer(tgen, dut, iif, STAR, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r1: Verify (*, G) ip mroutes")
+    result = verify_mroutes(tgen, dut, STAR, GROUP_ADDRESS_1, iif, oif)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r1: Verify (S, G) ip mroutes")
+    iif = TOPO["routers"]["r1"]["links"]["r3"]["interface"]
+    SOURCE_ADDRESS = TOPO["routers"]["r5"]["links"]["r3"]["ipv6"].split("/")[0]
+    result = verify_mroutes(tgen, dut, SOURCE_ADDRESS, GROUP_ADDRESS_1, iif, oif)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r1: Verify (S, G) upstream IIF interface")
+    result = verify_upstream_iif(tgen, dut, iif, SOURCE_ADDRESS, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r1: Verify (S, G) upstream join state and join timer")
+    result = verify_join_state_and_timer(
+        tgen, dut, iif, SOURCE_ADDRESS, GROUP_ADDRESS_1
+    )
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r2: Verify (*, G) upstream IIF interface")
+    dut = "r2"
+    iif = "lo"
+    result = verify_upstream_iif(tgen, dut, iif, STAR, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
+
+    step("r2: Verify (*, G) upstream join state and join timer")
+    result = verify_join_state_and_timer(tgen, dut, iif, STAR, GROUP_ADDRESS_1)
+    assert result is True, "Testcase {} :Failed \n Error: {}".format(tc_name, result)
```

Comment:
nit: I see this `Testcase {} :Failed \n Error: {}` everywhere, probably would be good to reuse this pattern overall.

Reply:
Done